frontend_tests/node_tests/popover_menus_data.js

```diff
@@ -0,0 +1,252 @@
+    assert.equal(response.should_display_delete_option, true);
+    assert.equal(response.should_display_read_receipts_option, true);
+    assert.equal(response.should_display_reminder_option, false);
+    assert.equal(response.should_display_quote_and_reply, true);
+});
+
+test("not_my_message_view_actions", () => {
+    set_page_params_no_edit_restrictions();
+    // Get message that is only viewable
+
+    const list = init_message_list();
+    message_lists.set_current(list);
+
+    // Message is sent by somebody else and is a stream with previous history.
+    // I should only be able to view this message with no edit/move permissions.
+    const messages = [
+        {
+            id: 1,
+            sender_id: mike.user_id,
+            is_hidden: false,
+            sent_by_me: false,
+            locally_echoed: false,
+            is_stream: true,
+            stream_id: 1,
+            collapsed: false,
+            unread: false,
+            edit_history: [
+                {
+                    prev_content: "Previous content",
+                    prev_stream: 0,
+                    prev_topic: "Previous topic",
+                },
+            ],
+        },
+    ];
+
+    add_message_with_view(list, messages);
+
+    const response = popover_menus_data.get_actions_popover_content_context(1);
+
+    assert.equal(response.view_source_menu_item, "translated: View message source");
+    assert.equal(response.editability_menu_item, undefined);
+    assert.equal(response.move_message_menu_item, undefined);
+});
+
+test("not_my_message_view_source_and_move", () => {
+    set_page_params_no_edit_restrictions();
+
+    // Get message that is movable with viewable source
+
+    const list = init_message_list();
+    message_lists.set_current(list);
+
+    // Message tests edge case where message is sent by someone else.
+    // Message is movable, however--I should have only view permissions with the exception of moving the message.
+    const messages = [
+        {
+            id: 1,
+            sender_id: mike.user_id,
+            is_hidden: false,
+            sent_by_me: false,
+            locally_echoed: false,
+            is_stream: true,
+            stream_id: 1,
+            type: "stream",
+            unread: false,
+            collapsed: false,
+            topic: compose.empty_topic_placeholder(),
+            edit_history: [
+                {
+                    prev_content: "Previous content",
+                    prev_stream: 0,
+                    prev_topic: "Previous topic",
+                },
+            ],
+        },
+    ];
+
+    add_message_with_view(list, messages);
+
+    const response = popover_menus_data.get_actions_popover_content_context(1);
+    assert.equal(response.view_source_menu_item, "translated: View message source");
+    assert.equal(response.editability_menu_item, undefined);
+    assert.equal(response.move_message_menu_item, "translated: Move messages");
+});
```

Comment:
Important tests that are missing from this are tests for different kinds of users in all the above 3 cases.

Reply:
Hey @amanagr , could you please elaborate the fact about different kinds of users, is there any documentation I can refer to understand different kinds of users with different permissions, Thanks.

Reply:
@palashb01 see https://zulip.com/help/roles-and-permissions

Reply:
Mostly I am looking for `admins, moderators, members, guests and spectators`.

Reply:
Got it , working on it.

Reply:
Hey @amanagr , I have changed the first test case according to different users, could you please verify whether the implementation is right or some changes are required, will change the other two cases after this, Thanks.